### ISSUE TITLE
subdomain whitelist

### ISSUE THREAD

**Issue by jimdigriz** [raicem/impulse-blocker#63]
Great tool but the one problem I regularly run into is my youtube.com listing.

The result is that if I enter my username and password into any Google login prompt, I get dropped onto:
```
Impulse Blocker
Your impulse to visit accounts.youtube.com is successfully blocked!
```

Google seem to use this as to set some authentication cookie for me, I have no idea why they do this (probably ads, duh!) but it is what it is.

Is there a way for me to allow 'accounts.youtube.com' whilst blocking everything else?

**Comment by jardondiego** [comment]
I run into the same problem constantly so it would be nice to work on this. Let's not forget that we can contribute!

**Comment by jimdigriz** [comment]
My advice is to treat the `webRequest` filter as a first-pass filter that only triggers your callback function when there is something of interest. This filter covers 99%+ of requests and means your extension only sees the traffic it needs to actually make a decision on.

The hook looks to be at https://github.com/raicem/impulse-blocker/blob/master/src/ImpulseBlocker.js#L62 and that hardcoded 'block' instruction will need to be replaced with a function that hooks in your rule engine.

Holding me back submitting a PR is that my opinion on how this should work (UI/UX wise) is probably different to how you see it and that decision is ultimately yours, once the UX/UI decision is the actual rule engine work is straight forward.

One recommendation though, consider using the https://publicsuffix.org/ list to create your `webRequest` match pattern filter.

**Comment by neuhaus** [comment]
I agree with @jimdigriz, the webRequest.RequestFilter will return a match for all blacklisted sites.
That is the right time to check for matches against a whiteList before blocking the request.
